public/tier0/tslist.h

```diff
@@ -1,1027 +1,12 @@
-// FIFO from http://perso.wanadoo.fr/gmem/evenements/jim2002/articles/L17_Fober.pdf
-//
-//=============================================================================
-
-#ifndef TSLIST_H
-#define TSLIST_H
-
-#if defined( _WIN32 )
 #pragma once
-// Suppress this spurious warning:
-// warning C4700: uninitialized local variable 'oldHead' used
-#pragma warning( push )
-#pragma warning( disable : 4700 )
-#endif
 
-#if defined( USE_NATIVE_SLIST ) && !defined( _X360 )
-#define WIN32_LEAN_AND_MEAN
-#include <windows.h>
-#endif
-
-#include "tier0/dbg.h"
-#include "tier0/threadtools.h"
-#include "tier0/memalloc.h"
-#include "tier0/memdbgoff.h"
-
-#if defined( _X360 )
-#define USE_NATIVE_SLIST
-#endif
-
-//-----------------------------------------------------------------------------
-
-#if defined( PLATFORM_64BITS ) && !defined( PLATFORM_E2K )
-
-#if defined (PLATFORM_WINDOWS)
-//typedef __m128i int128;
-//inline int128 int128_zero()	{ return _mm_setzero_si128(); }
-#else  // PLATFORM_WINDOWS
-typedef __int128_t int128;
-#define int128_zero() 0
-#endif// PLATFORM_WINDOWS
-
-#define TSLIST_HEAD_ALIGNMENT 16
-#define TSLIST_NODE_ALIGNMENT 16
-
-#ifdef POSIX
-inline bool ThreadInterlockedAssignIf128( int128 volatile * pDest, const int128 &value, const int128 &comparand ) 
-{
-    // We do not want the original comparand modified by the swap
-    // so operate on a local copy.
-    int128 local_comparand = comparand;
-	return __sync_bool_compare_and_swap( pDest, local_comparand, value );
-}
-#endif
+#include "platform.h"
 
-inline bool ThreadInterlockedAssignIf64x128( volatile int128 *pDest, const int128 &value, const int128 &comperand )
-{
-	return ThreadInterlockedAssignIf128( pDest, value, comperand );
-}
+#if defined(PLATFORM_E2K)
+#include "tslist_alternative.h"
 #else
-#define TSLIST_HEAD_ALIGNMENT 8
-#define TSLIST_NODE_ALIGNMENT 8
-inline bool ThreadInterlockedAssignIf64x128( volatile int64 *pDest, const int64 value, const int64 comperand )
-{
-	return ThreadInterlockedAssignIf64( pDest, value, comperand );
-}
+#include "tslist_atomics.h"
 #endif
 
-#ifdef _MSC_VER
-#define TSLIST_HEAD_ALIGN DECL_ALIGN(TSLIST_HEAD_ALIGNMENT)
-#define TSLIST_NODE_ALIGN DECL_ALIGN(TSLIST_NODE_ALIGNMENT)
-#define TSLIST_HEAD_ALIGN_POST
-#define TSLIST_NODE_ALIGN_POST
-#elif defined( GNUC )
-#define TSLIST_HEAD_ALIGN 
-#define TSLIST_NODE_ALIGN 
-#define TSLIST_HEAD_ALIGN_POST DECL_ALIGN(TSLIST_HEAD_ALIGNMENT)
-#define TSLIST_NODE_ALIGN_POST DECL_ALIGN(TSLIST_NODE_ALIGNMENT)
-#elif defined( _PS3 )
-#define TSLIST_HEAD_ALIGNMENT 8
-#define TSLIST_NODE_ALIGNMENT 8
-
-#define TSLIST_HEAD_ALIGN ALIGN8
-#define TSLIST_NODE_ALIGN ALIGN8
-#define TSLIST_HEAD_ALIGN_POST ALIGN8_POST
-#define TSLIST_NODE_ALIGN_POST ALIGN8_POST
-
-#else
-#error
-#endif
-
-//-----------------------------------------------------------------------------
-
 PLATFORM_INTERFACE bool RunTSQueueTests( int nListSize = 10000, int nTests = 1 );
```

Comment:
these 2 lines need to go in your tslist_alternative file, it's redefined in tslists_atomics.h

Reply:
Fixed